tasks/simplification.py

```diff
@@ -0,0 +1,142 @@
+from util import PostgresTarget, shell
+from meta import CurrentSession
+
+OBSERVATORY_SCHEMA = 'observatory'
+DEFAULT_GEOMFIELD = 'the_geom'
+TMP_DIRECTORY = 'tmp'
+SIMPLIFICATION_DIRECTORY = 'simplification'
+SIMPLIFIED_SUFFIX = '_simpl'
+SKIPFAILURES_NO = 'no'  # Avoids https://trac.osgeo.org/gdal/ticket/6803
+SKIPFAILURES_YES = 'yes'
+DEFAULT_M_RETAIN_PERCENTAGE = '50'  # [0-100] Percentage of removable vertices retained by mapshaper
+DEFAULT_P_RETAIN_FACTOR = '50'  # Retain factor used for PostGIS simplification (this is NOT a percentage) \
+# The higher the retain factor, the lower the simplification
+DEFAULT_MAX_MEMORY = '8192'
+
+
+def tmp_directory(schema, table):
+    return os.path.join(TMP_DIRECTORY, SIMPLIFICATION_DIRECTORY,
+                        '{schema}.{table}'.format(schema=schema, table=table))
+
+
+def shp_filename(table, suffix=''):
+    return '{table}{suffix}.shp'.format(table=table, suffix=suffix)
+
+
+class ExportShapefile(Task):
+    schema = Parameter()
+    table = Parameter()
+    skipfailures = Parameter(default=SKIPFAILURES_NO)
+
+    def run(self):
+        self.output().makedirs()
+        cmd = 'ogr2ogr -f "ESRI Shapefile" {shapefile} ' \
+              '{skipfailures} ' \
+              '"PG:dbname=$PGDATABASE active_schema={schema}" {table} '.format(
+                  shapefile=self.output().path, schema=self.schema, table=self.table,
+                  skipfailures='-skipfailures' if self.skipfailures.lower() == SKIPFAILURES_YES else '')
+        shell(cmd)
+
+    def output(self):
+        return LocalTarget(os.path.join(tmp_directory(self.schema, self.table),
+                                        shp_filename(self.table)))
+
+
+class SimplifyShapefile(Task):
+    schema = Parameter()
+    table_input = Parameter()
+    table_output = Parameter()
+    retainpercentage = Parameter(default=DEFAULT_M_RETAIN_PERCENTAGE)
+    skipfailures = Parameter(default=SKIPFAILURES_NO)
+    maxmemory = Parameter(default=DEFAULT_MAX_MEMORY)
+
+    def requires(self):
+        return ExportShapefile(schema=self.schema, table=self.table_input, skipfailures=self.skipfailures)
+
+    def run(self):
+        cmd = 'node --max-old-space-size={maxmemory} `which mapshaper` ' \
+              '{input} snap -simplify {retainpercentage}% keep-shapes -o {output}'.format(
+                maxmemory=self.maxmemory,
+                input=os.path.join(tmp_directory(self.schema, self.table_input),
+                                   shp_filename(self.table_input)),
+                retainpercentage=self.retainpercentage,
+                output=self.output().path)
+        shell(cmd)
+
+    def output(self):
+        return LocalTarget(os.path.join(tmp_directory(self.schema, self.table_input),
+                                        shp_filename(self.table_output)))
+
+
+class SimplifyGeometriesMapshaper(Task):
+    schema = Parameter()
+    table_input = Parameter()
+    table_output = Parameter()
```

Comment:
`table_output` should probably be decided by this task. Tasks that call this one can check this task output to determine where the destination table is.

Maybe even just making this task inherit from `TempTableTask` which automatically names tables is enough (although that might not work since this task is in a different file than the rest of the import process).

Reply:
I would prefer keeping the `table_output` parameter because it allows us to execute the simplify task separately (for example from the command line) maintaining full control of the output table

Reply:
Uhm, fair point. What about adding a default then? That way we support both usecases. 

If this sounds good, give it a try. But only if it's easy to do, if there are problems with making one parameter default to depend on another value, let's leave it as it is.

Reply:
Done with a bit of magic (rewriting the `__init__` method) but the solution is effective and not _very_ ugly